src/aleph/vm/controllers/firecracker/executable.py

```diff
@@ -306,8 +306,8 @@ async def start_guest_api(self):
         logger.debug(f"started guest API for {self.vm_id}")
 
     async def stop_guest_api(self):
-        if self.guest_api_process and self.guest_api_process._popen:
-            self.guest_api_process.terminate()
+        if self.guest_api_process and self.guest_api_process.is_alive():
+            self.guest_api_process.join(timeout=10)
```

Comment:
Why remove `.terminate()` ?

According to [the Python documentation](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.Pool.join):

> `join()`: Wait for the worker processes to exit. One must call [close()](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.Pool.close) or [terminate()](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.Pool.terminate) before using [join()](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.Pool.join).